File: scratch/test_lynk_api.php
<?php

require __DIR__ . '/../vendor/autoload.php';

// Test Lynk API or Next.js data endpoint
$urls = [
    'https://lynk.id/api/v1/link/Pv23p2E',
    'https://lynk.id/api/link/Pv23p2E',
    'https://lynk.id/api/p/Pv23p2E',
    'https://lynk.id/api/product/Pv23p2E',
];

foreach ($urls as $u) {
    $ch = curl_init();
    curl_setopt($ch, CURLOPT_URL, $u);
    curl_setopt($ch, CURLOPT_RETURNTRANSFER, true);
    curl_setopt($ch, CURLOPT_SSL_VERIFYPEER, false);
    curl_setopt($ch, CURLOPT_HTTPHEADER, [
        'User-Agent: Mozilla/5.0 (Windows NT 10.0; Win64; x64)',
        'Accept: application/json'
    ]);
    $res = curl_exec($ch);
    $code = curl_getinfo($ch, CURLINFO_HTTP_CODE);
    curl_close($ch);
    echo "$u => Status: $code, Response: " . substr($res, 0, 200) . "\n";
}

File: scratch/test_lynk_api.py
import urllib.request
import json

hashid = "Pv23p2E"
user = "mindiw"

# Test endpoints
urls = [
    f"https://lynk.id/v1/product/{hashid}",
    f"https://lynk.id/v1/products/{hashid}",
    f"https://lynk.id/v1/product/detail/{hashid}",
    f"https://lynk.id/v1/product/desc/{hashid}",
    f"https://lynk.id/v1/content/product/{hashid}",
    f"https://lynk.id/v1/store/product/{hashid}",
    f"https://lynk.id/v1/creator/{user}/product/{hashid}",
    f"https://lynk.id/v1/page/product/{hashid}",
    f"https://lynk.id/v1/description?hashid={hashid}",
]

headers = {
    'User-Agent': 'Mozilla/5.0 (Windows NT 10.0; Win64; x64) AppleWebKit/537.36 (KHTML, like Gecko) Chrome/125.0.0.0 Safari/537.36',
    'Accept': 'application/json, text/plain, */*',
    'X-Requested-With': 'XMLHttpRequest',
    'Referer': f'https://lynk.id/{user}/{hashid}'
}

for u in urls:
    try:
        req = urllib.request.Request(u, headers=headers)
        res = urllib.request.urlopen(req, timeout=4)
        body = res.read().decode('utf-8')
        print(f"SUCCESS GET {u} -> Status: {res.status}, Len: {len(body)}")
        print(f"Body: {body[:300]}")
    except Exception as e:
        print(f"FAIL GET {u} -> {e}")

# Also test POST
post_data = json.dumps({"hashid": hashid, "username": user}).encode('utf-8')
post_urls = [
    f"https://lynk.id/v1/product/description",
    f"https://lynk.id/v1/product/detail",
    f"https://lynk.id/v1/product/get",
    f"https://lynk.id/v1/content/get",
]

for u in post_urls:
    try:
        req = urllib.request.Request(u, data=post_data, headers={**headers, 'Content-Type': 'application/json'})
        res = urllib.request.urlopen(req, timeout=4)
        body = res.read().decode('utf-8')
        print(f"SUCCESS POST {u} -> Status: {res.status}, Len: {len(body)}")
        print(f"Body: {body[:300]}")
    except Exception as e:
        print(f"FAIL POST {u} -> {e}")
